### ISSUE TITLE
Different Name for Namespace in EarlyBoundGenerator

### ISSUE THREAD

**Issue by craigseymour** [daryllabar/DLaB.Xrm.XrmToolBoxTools#54]
I'd like to be able to specify different namespaces for Entities and OptionSets 
e.g. 

- MyCompany.Crm.Entities
- MyCompany.Crm.OptionSets

At the moment, they are set the same.

**Comment by daryllabar** [comment]
So would this mean creating an Entity/OptionSet/Action specific Namespace box, and if they are different, adding them as references to all files correct?

**Comment by craigseymour** [comment]
Yep - I think would cover it.

**Comment by daryllabar** [comment]
I'll look to adding additional namespaces as well for the XRM namespaces...  It makes the generated code rather hard to read at times...

**Comment by craigseymour** [comment]
To what would the XRM namespaces applied?

**Comment by daryllabar** [comment]
Let me redefine.  I would add some new using statements, so that everything doesn't have to be explicitly defined.  You'd only ever care if you looked at the actual generated code.  Make sense?

**Comment by craigseymour** [comment]
Gotcha.

**Comment by daryllabar** [comment]
Is this something that is still desired?

**Comment by marcscheuner-bfh** [comment]
Yes - definitely still desired - either have separate namespace boxes for entities, optionsets etc. - or have a Yes/No option saying "Add relative path to namespace" - when I define a namespace of "MyApp.Core", and a relative folder for the entities of "Entities" and for option sets of "OptionSets", I'd also like to be able to add these relative folder names to the namespace, e.g. "MyApp.Core.Entities" and "MyApp.Core.OptionSets"

**Comment by daryllabar** [comment]
@marcscheuner-bfh  how would you propose handling local option sets?  By default these are generated in the same file as the Entity.  I'm assuming in that case, you would prefer to have the two different namespaces in the same file:

```cs 
// Single file:
namespace X.OptionSets {
    // Local Option Sets
}

namespace X.Entities {
    // Entity Class
}
```

**Comment by marcscheuner-bfh** [comment]
> @marcscheuner-bfh how would you propose handling local option sets? By default these are generated in the same file as the Entity. I'm assuming in that case, you would prefer to have the two different namespaces in the same file:

Yes - if you desire to have separate namespaces for entities and option sets, I'd also expect those to be applied even inside a single file.
